src/main/java/bio/terra/service/dataset/flight/inheritsteward/GetSnapshotIdsStep.java

```diff
@@ -8,19 +11,47 @@
 import bio.terra.stairway.exception.RetryException;
 import java.util.List;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 public class GetSnapshotIdsStep extends DefaultUndoStep {
   private final SnapshotDao snapshotDao;
+  private final IamService iamService;
+  private final AuthenticatedUserRequest userReq;
   private final UUID datasetId;
+  private final boolean inheritSteward;
 
-  public GetSnapshotIdsStep(SnapshotDao snapshotDao, UUID datasetId) {
+  public GetSnapshotIdsStep(
+      SnapshotDao snapshotDao,
+      IamService iamService,
+      AuthenticatedUserRequest userReq,
+      UUID datasetId,
+      boolean inheritSteward) {
     this.snapshotDao = snapshotDao;
+    this.iamService = iamService;
+    this.userReq = userReq;
     this.datasetId = datasetId;
+    this.inheritSteward = inheritSteward;
   }
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
-    List<UUID> snapshotIds = snapshotDao.getSnapshotIds(datasetId);
+    List<UUID> snapshotIds;
+    if (inheritSteward) {
+      snapshotIds = snapshotDao.getSnapshotIds(datasetId);
+    } else {
+      snapshotIds =
+          iamService
+              .listResourceChildren(
+                  userReq.getToken(), bio.terra.service.auth.iam.IamResourceType.DATASET, datasetId)
```

Comment:
Can you import `IamResourceType`?
```suggestion
                  userReq.getToken(), IamResourceType.DATASET, datasetId)
```